packages/lesswrong/components/common/SelectionButtonWrapper.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState, useRef, useCallback } from 'react';
+import { getVotingSystemByName } from '../../lib/voting/votingSystems';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import { useNamesAttachedReactionsVoting } from '../votes/NamesAttachedReactionsVoteOnComment';
+import { useVote } from '../votes/withVote';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  button: {
+    zIndex: 1000,
+    backgroundColor: theme.palette.background.paper,
+    borderRadius: 4,
+    boxShadow: theme.shadows[2],
+    padding: 4,
+    width: 100,
+    height: 100
+  }
+});
+
+export const SelectionButtonWrapper = ({classes, comment, children}: {
+  classes: ClassesType,
+  comment: CommentsList,
+  children: React.ReactNode,
+}) => {
+  const selectionRef = useRef<HTMLDivElement|null>(null);
+  const popupRef = useRef<HTMLDivElement|null>(null);
+  const [quote, setQuote] = useState<string>("");
+  const [anchorEl, setAnchorEl] = useState<HTMLElement|null>(null);
+
+  const { PopperCard, AddReactionButton, ReactionsPalette } = Components;
+
+  const votingSystemName = comment.votingSystem || "default";
+  const votingSystem = getVotingSystemByName(votingSystemName);
+  const voteProps = useVote(comment, "Comments", votingSystem);
+  
+  const { getCurrentUserReactionVote, toggleReaction } = useNamesAttachedReactionsVoting(voteProps);
+
+  // note: I originally just made this a function. ChatGPT claimed that it would
+  // result in a memory leak because it would create a new function every time
+  // the component re-renders. I don't know if that's true, but I'm using
+  // useCallback to memoize it just in case.
+  // ChatGPT also... generated this note, so watch out for that
```

Comment:
It's correct to have it wrapped in `useCallback` (for any substantial function in general), but not because of a memory leak. Whether there's a memory leak or not depends only on whether the `useEffect` below this cleans up after itself properly (it doe).